src/Logging.jl

```diff
@@ -67,25 +115,47 @@ show(io::IO, logger::Logger) = print(io, "Logger(", join([logger.name,
                                                           logger.output,
                                                           logger.parent.name], ","), ")")
 
-const _root = Logger("root", WARNING, STDERR)
-Logger(name::AbstractString;args...) = configure(Logger(name, WARNING, [STDERR], _root); args...)
+const _root = Logger("root", WARNING, LogIO(STDERR))
+Logger(name::AbstractString; kwargs...) = configure(Logger(name, WARNING, [LogIO(STDERR)], _root); kwargs...)
 Logger() = Logger("logger")
 
 write_log(syslog::SysLog, color::Symbol, msg::AbstractString) = send(syslog.socket, syslog.ip, syslog.port, length(msg) > syslog.maxlength ? msg[1:syslog.maxlength] : msg)
-write_log{T<:IO}(output::T, color::Symbol, msg::AbstractString) = (print(output, msg); flush(output))
-write_log(output::Base.TTY, color::Symbol, msg::AbstractString) = Base.print_with_color(color, output, msg)
-
-function log(syslog::SysLog, level::LogLevel, color::Symbol, logger_name::AbstractString, msg...)
-    # syslog needs a timestamp in the form: YYYY-MM-DDTHH:MM:SS-TZ:TZ
-    t = time()
-    timestamp = string(Libc.strftime("%Y-%m-%dT%H:%M:%S",t), Libc.strftime("%z",t)[1:end-2], ":", Libc.strftime("%z",t)[end-1:end])
-    logstring = string("<", (UInt16(syslog.facility) << 3) + UInt16(level), ">1 ", timestamp, " ", syslog.machine, " ", syslog.user, " - - - ", level, ":", logger_name,":", msg...)
-    write_log(syslog, color, logstring)
+write_log{T<:IO}(output::T, color::Symbol, msg::AbstractString) = (println(output, msg); flush(output))
+write_log(output::Base.TTY, color::Symbol, msg::AbstractString) = (Base.println_with_color(color, output, msg); flush(output))
+
+function formatlog(format::AbstractString, logging_fields::Dict)
+    function substr(s)
+        v = get(logging_fields, s[3:end-1], "<none>")
+        return isa(v, Function) ? string(v()) : string(v)
+    end
+    return replace(format, r"%\([^)]+\)", substr)
```

Comment:
This might be worth breaking out to a `const` so that we can name it and make it apparent what this regex is.